File: rb/features.rb
# JokeFather SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module JokeFatherFeatures
  def self.make_feature(name)
    case name
    when "base"
      JokeFatherBaseFeature.new
    when "test"
      JokeFatherTestFeature.new
    else
      JokeFatherBaseFeature.new
    end
  end
end

File: py/features.py
# JokeFather SDK feature factory

from feature.base_feature import JokeFatherBaseFeature
from feature.test_feature import JokeFatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JokeFatherBaseFeature(),
        "test": lambda: JokeFatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
